server/tests/api_v4/authenticated/test_shamir_recovery_setup.py

```diff
@@ -0,0 +1,91 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import pytest
+
+from parsec._parsec import (
+    DateTime,
+    ShamirRecoveryBriefCertificate,
+    ShamirRecoveryShareCertificate,
+    ShamirRevealToken,
+    authenticated_cmds,
+)
+from tests.common import CoolorgRpcClients
+from tests.common.client import setup_shamir_for_coolorg
+
+
+async def test_authenticated_shamir_recovery_setup_ok(
+    coolorg: CoolorgRpcClients, with_postgresql: bool
+) -> None:
+    if with_postgresql:
+        pytest.skip("TODO: postgre not implemented yet")
+    share = ShamirRecoveryShareCertificate(
+        coolorg.alice.device_id, DateTime.now(), coolorg.mallory.user_id, b"abc"
+    )
+    brief = ShamirRecoveryBriefCertificate(
+        author=coolorg.alice.device_id,
+        timestamp=DateTime.now(),
+        threshold=1,
+        per_recipient_shares={coolorg.mallory.user_id: 2},
+    )
+
+    setup = authenticated_cmds.v4.shamir_recovery_setup.ShamirRecoverySetup(
+        b"abc",
+        ShamirRevealToken.new(),
+        brief.dump_and_sign(coolorg.alice.signing_key),
+        [share.dump_and_sign(coolorg.alice.signing_key)],
+    )
+    rep = await coolorg.alice.shamir_recovery_setup(setup)
+    assert rep == authenticated_cmds.v4.shamir_recovery_setup.RepOk()
+
+    # TODO dump
+
+
+async def test_authenticated_shamir_recovery_setup_already_set(
+    coolorg: CoolorgRpcClients, with_postgresql: bool
+) -> None:
+    # Setup previous shamir
+    if with_postgresql:
+        pytest.skip("TODO: postgre not implemented yet")
+    share = ShamirRecoveryShareCertificate(
+        coolorg.alice.device_id, DateTime.now(), coolorg.mallory.user_id, b"abc"
+    )
+    brief = ShamirRecoveryBriefCertificate(
+        author=coolorg.alice.device_id,
+        timestamp=DateTime.now(),
+        threshold=1,
+        per_recipient_shares={coolorg.mallory.user_id: 2},
+    )
+
+    await setup_shamir_for_coolorg(coolorg)
+
+    # attempt to overwrite setup
+    setup = authenticated_cmds.v4.shamir_recovery_setup.ShamirRecoverySetup(
+        b"def",
+        ShamirRevealToken.new(),
+        brief.dump_and_sign(coolorg.alice.signing_key),
+        [share.dump_and_sign(coolorg.alice.signing_key)],
+    )
+    rep = await coolorg.alice.shamir_recovery_setup(setup)
+    assert rep == authenticated_cmds.v4.shamir_recovery_setup.RepAlreadySet()
+
+
+@pytest.mark.xfail(reason="TODO")
```

Comment:
I wasn't able to find the equivalent in v2.17.0, so I don't know when to return the InvalidCertification errror

Reply:
That error is when the provided certificate is invalid, the certificate is provided as `bytes` but it should be validated by the server (or another client)

We have a similar example:

https://github.com/Scille/parsec-cloud/blob/771699834edb838f0e1d6fc8b72f65a20da802ef/server/tests/api_v4/authenticated/test_realm_create.py#L82